src/server/page_server.cpp

```diff
@@ -107,64 +109,151 @@ page_server::connection_handler::receive_log_prior_list (tran_server_conn_t::seq
   log_Gl.get_log_prior_receiver ().push_message (std::move (a_ip.pull_payload ()));
 }
 
+template<class F>
 void
-page_server::connection_handler::receive_log_page_fetch (tran_server_conn_t::sequenced_payload &a_sp)
+page_server::connection_handler::push_async_response (F &&a_func, tran_server_conn_t::sequenced_payload &&a_sp)
+{
+  m_ps.get_responder ().async_execute (std::ref (*m_conn), std::move (a_sp),
+				       std::bind (std::forward<F> (a_func), this,
+					   std::placeholders::_1, std::placeholders::_2));
+}
+
+void
+page_server::connection_handler::fetch_log_page (cubthread::entry &context, std::string &a_payload)
```

Comment:
I see nothing in this function that mandates it being a member of the connection handler. It can be moved to `log_page_buffer`.
If there is ever a need to have member functions as handler functions, `push_async_response` can have two flavors, one that binds a member function; one that maps a global function.